src/app/modules/submission/submission.types.ts

```diff
@@ -1,18 +1,37 @@
+import { Opaque } from 'type-fest'
+
 import {
   IAttachmentResponse,
   ICheckboxResponse,
   ISingleAnswerResponse,
   ITableResponse,
 } from 'src/types/response'
 
-import { BasicField } from '../../../types/field'
+import { FieldIdSet } from '../../../shared/util/logic'
+import { BasicField, IFieldSchema } from '../../../types/field'
 
 export type ProcessedResponse = {
   question: string
   isVisible?: boolean
   isUserVerified?: boolean
 }
 
+/**
+ * Represents a field map that is guaranteed to contain the id of
+ * ALL responses in an incoming response.
```

Comment:
I think it might be clearer to say "the id of ALL **field** responses in an incoming response"?